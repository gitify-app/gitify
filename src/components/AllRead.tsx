import { useMemo } from 'react';
import { Constants } from '../utils/constants';

export const AllRead = () => {
  const emoji = useMemo(
    () =>
      Constants.ALL_READ_EMOJIS[
        Math.floor(Math.random() * Constants.ALL_READ_EMOJIS.length)
      ],
    [],
  );

  return (
    <div className="flex flex-1 flex-col items-center justify-center bg-white p-4 text-black dark:bg-gray-dark dark:text-white">
      <h1 className="mb-5 text-5xl">{emoji}</h1>

      <h2 className="mb-2 text-xl font-semibold">No new notifications.</h2>
    </div>
  );
};
