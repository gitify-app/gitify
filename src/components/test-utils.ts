const originalMathRandom = Math.random;

export const mockMathRandom = (mockValue: number) => {
  beforeEach(() => {
    global.Math.random = jest.fn(() => mockValue);
  });

  afterEach(() => {
    global.Math.random = originalMathRandom;
  });
};
