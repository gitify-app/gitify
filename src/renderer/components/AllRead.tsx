import { type FC, useContext, useMemo } from 'react';

import { AppContext } from '../context/App';
import { Constants } from '../utils/constants';
import { hasAnyFiltersSet } from '../utils/filters';
import { EmojiSplash } from './layout/EmojiSplash';

interface IAllRead {
  fullHeight?: boolean;
}

export const AllRead: FC<IAllRead> = ({ fullHeight = true }: IAllRead) => {
  const { settings } = useContext(AppContext);

  const hasFilters = hasAnyFiltersSet(settings);

  const emoji = useMemo(
    () =>
      Constants.ALL_READ_EMOJIS[
        Math.floor(Math.random() * Constants.ALL_READ_EMOJIS.length)
      ],
    [],
  );

  const heading = `No new ${hasFilters ? 'filtered ' : ''} notifications`;

  return (
    <EmojiSplash emoji={emoji} heading={heading} fullHeight={fullHeight} />
  );
};
