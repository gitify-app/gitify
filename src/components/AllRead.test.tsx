import * as TestRenderer from 'react-test-renderer';

import { mockMathRandom } from './test-utils';

import { AllRead } from './AllRead';

describe('components/AllRead.tsx', () => {
  // The read emoji randomly rotates, but then the snapshots would never match
  // Have to make it consistent so the emojis are always the same
  mockMathRandom(0.1);

  it('should render itself & its children', () => {
    const tree = TestRenderer.create(<AllRead />);

    expect(tree).toMatchSnapshot();
  });
});
