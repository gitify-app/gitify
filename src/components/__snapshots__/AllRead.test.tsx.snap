// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AllRead.tsx should render itself & its children 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-1 flex-col items-center justify-center bg-white p-4 text-black dark:bg-gray-dark dark:text-white"
      >
        <h1
          class="mb-5 text-5xl"
        >
          ðŸ˜‰
        </h1>
        <h2
          class="mb-2 text-xl font-semibold"
        >
          No new notifications.
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-1 flex-col items-center justify-center bg-white p-4 text-black dark:bg-gray-dark dark:text-white"
    >
      <h1
        class="mb-5 text-5xl"
      >
        ðŸ˜‰
      </h1>
      <h2
        class="mb-2 text-xl font-semibold"
      >
        No new notifications.
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
